darts/dataprocessing/utils.py

```diff
@@ -0,0 +1,149 @@
+"""
+Data Processing Utils
+---------------------
+"""
+import numpy as np
+
+from typing import TypeVar, Callable, Optional, Tuple, List, Any, Dict, Type
+
+from darts import TimeSeries
+from darts.logging import raise_if, get_logger
+from darts.dataprocessing.transformers import BaseDataTransformer, InvertibleDataTransformer, FittableDataTransformer
+
+logger = get_logger(__name__)
+T = TypeVar('T')
+_callable_with_args_kwargs = Callable[[T, Tuple[Any, ...], Dict[str, any]], T]
+
+
+def _create_wrapper_data_transformer(parent_classes: List[Type[BaseDataTransformer[T]]],
+                                     transform: _callable_with_args_kwargs,
+                                     inverse_transform: Optional[_callable_with_args_kwargs] = None,
+                                     fit: Optional[_callable_with_args_kwargs] = None,
+                                     name: str = "WrapperDataTransformer"):
+    class _WrapperDataTransformer(*parent_classes):
+        def __init__(self,
+                     transform: _callable_with_args_kwargs,
+                     inverse_transform: Optional[_callable_with_args_kwargs] = None,
+                     fit: Optional[_callable_with_args_kwargs] = None,
+                     name: str = "WrapperDataTransformer"):
+            raise_if(transform is None, "transform cannot be None", logger)
+            raise_if(InvertibleDataTransformer in parent_classes and inverse_transform is None,
+                     "WrapperDataTransformer marked as invertible but no inverse_transform() function was provided",
+                     logger)
+            raise_if(FittableDataTransformer in parent_classes and fit is None,
+                     "WrapperDataTransfomer marked as fittable but no fit() function was provided")
+
+            super().__init__(name=name, validators=None)  # TODO: Add possibility to specify validators ?
+
+            self._transform = transform
+            self._inverse_transform = inverse_transform
+            self._fit = fit
+
+            self._name = name
+
+        def transform(self, data: T, *args, **kwargs) -> T:
+            return self._transform(data, *args, **kwargs)
+
+        if InvertibleDataTransformer in parent_classes:
+            def inverse_transform(self, data: T, *args, **kwargs) -> T:
+                return self._inverse_transform(data, *args, **kwargs)
+
+        if FittableDataTransformer in parent_classes:
+            def fit(self, data: T) -> 'Type[BaseDataTransformer[T]]':
+                self._fit(data)
```

Comment:
I think you're correct. These might not actually be very useful (except maybe for writing tests). I'll remove them and adapt the tests accordingly

Reply:
Something occured to me over the weekend looking at common transforms [here](https://otexts.com/fpp2/transformations.html). These transforms don't need to store state (and thus don't need a `fit()` method) and still would be very useful.

This made me realize a few things:
1. Sateless transformers performing basically the same function as a call to `map()` can be very useful. This way, such a call could be chained inside of a `Pipeline`, and provided that the function used with `map()` has an inverse, could also be easily inverted as part of a pipeline.
2. It could be very useful to be able to `map()` a `TimeSeries` with a `function(datapoint_timestamp, datapoint)` instead of only `function(datapoint)`.
3. For multivariate `TimeSeries`, we probably also want to add some way of specifying a subset of target series to apply a transform to, in order to be able to create more complex pipelines. It would however complexify them a fair bit, and make implementing one's own transformers more error prone. I can think of a couple ways to go about this, I think it might be worth it to discuss it offline.

To implement 1., since`WrapperDataTransformer` was quite convoluted and we only need a non-invertible and an invertible wrapper, my proposition is to add two classes instead, one that derives from `BaseTransformer` and one that derives from `InvertibleTransformer`. I'm not sure about how to name them yet, what do you say about `FunctionWrapper` and `InvertibleFunctionWrapper` ?